src/components/NavBar/ChainSelector.tsx

```diff
@@ -29,12 +31,42 @@ const NETWORK_SELECTOR_CHAINS = [
   SupportedChainId.BNB,
 ]
 
+if (!isProductionEnv()) {
+  NETWORK_SELECTOR_CHAINS.push(SupportedChainId.GOERLI)
+}
+
 interface ChainSelectorProps {
   leftAlign?: boolean
 }
 
+// accounts is an array of strings in the format of "eip155:<chainId>:<address>"
+function getChainsFromEIP155Accounts(accounts?: string[]): SupportedChainId[] {
+  if (!accounts) return []
+  return accounts.map((account) => {
+    const splitAccount = account.split(':')
+    return parseInt(splitAccount[1])
```

Comment:
i'm surprised TS lets you do this - console tells me `parseInt(undefined) = NaN`

Reply:
good catch. This is technically safe is we assume WC implementations are stable, but the types as written shouldn't allow this imo.